src/serialization.cpp

```diff
@@ -52,6 +52,17 @@ void zerr(int ret)
     }
 }
 
+// Make sure that z is deleted in case of exception
+template <int (*F)(z_stream*)>
+class ZlibAutoDeleter {
+public:
+	ZlibAutoDeleter(z_stream *ptr) : ptr_(ptr) {}
+	~ZlibAutoDeleter() { F(ptr_); }
+
+private:
+	z_stream *ptr_;
```

Comment:
(this is equivalent)
```suggestion
class ZlibAutoDeleter {
	z_stream *ptr;
public:
	ZlibAutoDeleter(z_stream *ptr) : ptr(ptr) {}
	~ZlibAutoDeleter() { F(ptr); }
```

Reply:
This is stupid an unnecessary. Why shouldn't I change it back?

Reply:
This is both one line shorter and clearer (no unnecessary `ptr` `ptr_` distinction)

Reply:
I don't consider one line shorter as a valid point. Follow code style and leave your sense of beauty for yourself. Thanks.

Reply:
Here are our code style guidelines, I don't see it mentioned: https://dev.minetest.net/Code_style_guidelines

> I don't consider one line shorter as a valid point.

What about the `ptr`, `ptr_` naming then?

Reply:
It was mentioned in the beginning. Minetest style guide doesn't say anything about private members naming, but google style guide says they should have trailing underscoreshttps://google.github.io/styleguide/cppguide.html#Variable_Names

Reply:
Well we don't use use Google's style guide. Searching through the source for `_;` I find that jsoncpp does this consistently and most other uses are in constructors but the other way around (as in `C::C(T *foo_) { foo = foo_; }`).
So I don't think new code should use a style that does not fit existing code.

Reply:
@savilli